relay-server/src/actors/envelopes.rs

```diff
@@ -228,16 +229,8 @@ impl EnvelopeManager {
         }
 
         // if we are in capture mode, we stash away the event instead of forwarding it.
-        if self.config.relay_mode() == RelayMode::Capture {
-            // XXX: this is wrong because captured_events does not take envelopes without
-            // event_id into account.
-            if let Some(event_id) = envelope.event_id() {
-                relay_log::debug!("capturing envelope");
-                self.captures.insert(event_id, Ok(envelope));
-            } else {
-                relay_log::debug!("dropping non event envelope");
-            }
-
+        if Capture::should_capture(&self.config) {
+            context.notify(Capture::accepted(envelope));
```

Comment:
Is it better to `.notify`, or to call `.handle()` directly?

Reply:
Good question, I went with notify just because that's how actix documents sending messages to the own actor. Probably calling handle would be more direct or even efficient. The immediate next PR is going to move this to another actor anyway, so calling `notify` here is closer to target behavior.